### DISCUSSION TITLE
Invasion Percolation and 2 component analysis

### DISCUSSION THREAD

**Discussion by ruchajk2597** [PMEAL/OpenPNM#2569] [Ideas]
Hello community, 

I am trying to run a flow simulation such that, I fill the pore network with one phase say water and then determine the  permeability of the second phase say air, nitrogen etc relative to this filled network. 
I found an example (https://openpnm.org/examples/applications/relative_permeability.html) on the openpnm website where percolation invasion is used. Unfortunately, this sample code throws a key error. 

My idea was to define a phase, apply a pressure  difference t fill the pore network. Using this new geometry define the 2nd phase and then check the permeability. 
Could you guide me if this would work or what the correct procedure or method wouldbe to carry out such a simulation?

Thank you in advance. 
Best Regards, 
Rucha Kunte

**Comment by jgostick** [comment]
Hi @ruchajk2597.  The example you found is for V3, which were are planning to release very soon, but I'm guessing that you are using V2?  You can find out by typing ``op.__version__`` at the python prompt.  Anyway, what you propose should work, more or less.  If you can't wait for the new version you can either get the new version using git, or you can checkout one of the older examples like [this one](https://github.com/PMEAL/OpenPNM/blob/release/examples/notebooks/algorithms/multiphase/relative_diffusivity.ipynb), which is for relative diffusion but the idea translates directly to flow too.

**Reply by ruchajk2597** [reply]
Hi, 
I am using version 2.6.1
When I run the Ordinary percolation algorithm, I get the following key error. 

KeyError                                  Traceback (most recent call last)
Input In [79], in <cell line: 4>()
      2 OP_1.setup(phase=water1, pore_volume='pore.volume', throat_volume='throat.volume')
      3 OP_1.set_inlets(pn.pores('xmin'))
----> 4 OP_1.run()

File C:\ProgramData\Anaconda3\lib\site-packages\openpnm\algorithms\OrdinaryPercolation.py:370, in OrdinaryPercolation.run(self, points, start, stop)
    367 # Parse inputs and generate list of invasion points if necessary
    368 if self.settings['mode'] == 'bond':
    369     self['throat.entry_pressure'] = \
--> 370         phase[self.settings['throat_entry_threshold']]
    371     if start is None:
    372         start = np.amin(self['throat.entry_pressure'])*0.5

File C:\ProgramData\Anaconda3\lib\site-packages\openpnm\phases\GenericPhase.py:99, in GenericPhase.__getitem__(self, key)
     93         mod = {'pore': mods.misc.from_neighbor_throats,
     94                'throat': mods.misc.from_neighbor_pores}
     95         self.add_model(propname=key,
     96                        model=mod[element],
     97                        prop=not_el + '.' + prop,
     98                        mode='mean')
---> 99 vals = super().__getitem__(key)
    100 return vals

File C:\ProgramData\Anaconda3\lib\site-packages\openpnm\core\Base.py:277, in Base.__getitem__(self, key)
    275     vals = super().__getitem__(key)
    276 else:
--> 277     raise KeyError(key)
    278 return vals

KeyError: 'throat.entry_pressure'


I understand this is because the parameter throat.entry_pressure is maybe called something else in my network? How can I fix this?


Looking forward to hearing from you. 
Thanks in advance. 

Best Regards, 
Rucha

**Comment by ruchajk2597** [comment]
Hi,

Thank you for the prompt reply.
The given example works for a randomly generated network. But in my case I am creating a network using the porespy snow algorithm.
It gives me the following error.


Something like this where my network generated does not have pore.diameter but pore.inscribed_diameter for example.
***@***.***

Could you please tell me how to fix this?

Thank you in advance.
Best Regards,
Rucha
Von: jgostick ***@***.***>
Gesendet: Montag, 19. September 2022 18:47
An: PMEAL/OpenPNM ***@***.***>
Cc: Kunte, Rucha ***@***.***>; Mention ***@***.***>
Betreff: Re: [PMEAL/OpenPNM] Invasion Percolation and 2 component analysis (Discussion #2569)


Hi @ruchajk2597<https://github.com/ruchajk2597>. The example you found is for V3, which were are planning to release very soon, but I'm guessing that you are using V2? You can find out by typing op.__version__ at the python prompt. Anyway, what you propose should work, more or less. If you can't wait for the new version you can either get the new version using git, or you can checkout one of the older examples like this one<https://github.com/PMEAL/OpenPNM/blob/release/examples/notebooks/algorithms/multiphase/relative_diffusivity.ipynb>, which is for relative diffusion but the idea translates directly to flow too.
